src/GafferSceneUI/TranslateTool.cpp

```diff
@@ -111,6 +112,32 @@ bool TranslateTool::affectsHandlesTransform( const Gaffer::Plug *input ) const
 		input == scenePlug()->transformPlug();
 }
 
+void TranslateTool::updateHandles()
+{
+	handles()->setTransform( handlesTransform() );
+
+	for( int i = 0; i < 3; ++i )
+	{
+		V3f handleDirection( 0 );
+		handleDirection[i] = 1.0f;
+		Translation translation = createTranslation( handleDirection );
+		bool editable = true;
+		for( int j = 0; j < 3; ++j )
+		{
+			if( translation.direction[j] != 0.0f )
+			{
+				const ValuePlug *plug = selection().transformPlug->translatePlug()->getChild( j );
```

Comment:
I've attempted to clarify this with an additional comment in 89ef0a8. Basically, locking a single plug can cause multiple handles to be disabled when the handles aren't aligned to the axes of the transform space (because of existing rotation). Were you just querying the code or have you seen problems in practice? I think it is possible that numerical imprecision in `createTranslation()` causes the translation values in some axes to be very small non-zero numbers rather than zero, and this could cause problems. If that's the case, I think the fix will need to be in `createTranslation()` rather than here.